gateway-provider-security-hadoopauth/src/main/java/org/apache/knox/gateway/hadoopauth/filter/HadoopAuthPostFilter.java

```diff
@@ -78,7 +79,14 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
     Subject subject = null;
     if (shouldUseJwtFilter(jwtFilter, (HttpServletRequest) request)) {
       try {
-        subject = jwtFilter.createSubjectFromToken(jwtFilter.getWireToken(request));
+        Pair<JWTFederationFilter.TokenType, String> wireToken = jwtFilter.getWireToken(request);
+        JWTFederationFilter.TokenType tokenType = wireToken.getLeft();
+        String token = wireToken.getRight();
+        if (JWTFederationFilter.TokenType.JWT.equals(tokenType)) {
+          subject = jwtFilter.createSubjectFromToken(token);
+        } else if (JWTFederationFilter.TokenType.Passcode.equals(tokenType)) {
```

Comment:
nit: this could be just `JWTFederationFilter.TokenType.PASSCODE.equals(tokenType)` for readability.

Reply:
I don't understand what you're suggesting here. Can you elaborate for me?

Reply:
I see, you mean `JWTFederationFilter.PASSCODE.equals(tokenType)` ; This would require changing the parsing logic and resulting Pair contents.